src/documents/serialisers.py

```diff
@@ -688,3 +690,98 @@ def _validate_task_id_list(self, tasks, name="tasks"):
     def validate_tasks(self, tasks):
         self._validate_task_id_list(tasks)
         return tasks
+
+
+class ObfuscatedPasswordField(serializers.Field):
+    """
+    Sends *** string instead of password in the clear
+    """
+
+    def to_representation(self, value):
+        return re.sub(".", "*", value)
+
+    def to_internal_value(self, data):
+        return data
+
+
+class MailAccountSerializer(serializers.ModelSerializer):
+    password = ObfuscatedPasswordField()
+
+    class Meta:
+        model = MailAccount
+        depth = 1
+        fields = [
+            "id",
+            "name",
+            "imap_server",
+            "imap_port",
+            "imap_security",
+            "username",
+            "password",
+            "character_set",
+        ]
+
+    def update(self, instance, validated_data):
+        if "password" in validated_data:
+            if len(validated_data.get("password").replace("*", "")) == 0:
+                validated_data.pop("password")
+        super().update(instance, validated_data)
+        return instance
+
+    def create(self, validated_data):
+        mail_account = MailAccount.objects.create(**validated_data)
+        return mail_account
+
+
+class AccountField(serializers.PrimaryKeyRelatedField):
+    def get_queryset(self):
+        return MailAccount.objects.all()
+
+
+class MailRuleSerializer(serializers.ModelSerializer):
```

Comment:
I'm not sure if this is the right place, but can we validate the inputs some?  Like a MOVE action with an empty parameter, or TAG with an empty parameter, etc

Reply:
Im not sure about this one, we get validation from the model and basic DRF stuff, do you think more is needed? Obviously please feel free to add here as Im not 100% sure what you mean

Reply:
Oh I gotcha now!